File: src/MisteryApp.Implementation/Services/UserService.cs
using MisteryApp.Abstractions.Interfaces;
using MisteryApp.Abstractions.Models;

namespace MisteryApp.Implementation.Services;

public class UserService(IUserRepository userRepository) : IUserService
{
    public virtual async Task<IReadOnlyList<User>> GetAllUsersAsync(CancellationToken cancellationToken)
    {
        return await userRepository.UserGetAllAsync(cancellationToken);
    }

    public virtual async Task<User?> GetUserByIdAsync(int id, CancellationToken cancellationToken)
    {
        return await userRepository.UserSingleOrDefaultByIdAsync(id, cancellationToken);
    }
}

File: src/misteryapp.web.client.react/src/services/user-service.ts
import type { IUserRepository } from '../domain/interfaces/i-user-repository'
import type { IUserService } from '../domain/interfaces/i-user-service'
import type { User } from '../domain/models/user'

export class UserService implements IUserService {
  constructor(private readonly userRepo: IUserRepository) {}

  getUsers(): Promise<readonly User[]> {
    return this.userRepo.userFindAll()
  }
}
